### PR TITLE
Set pre-net5 win-arm64 to x64 by default

### PR DESCRIPTION
**Pull request by sfoslund**
Addresses 3rd bullet of https://github.com/dotnet/sdk/issues/18832

### CODE REVIEW

**Comment by sfoslund** [comment]
@dsplaisted @wli3 is there a simple way to test this kind of change?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets [hunk 1]
```diff
@@ -137,6 +137,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DefaultAppHostRuntimeIdentifier Condition="$(DefaultAppHostRuntimeIdentifier.EndsWith('arm64')) and
                                   $(DefaultAppHostRuntimeIdentifier.StartsWith('osx')) and
                                   $([MSBuild]::VersionLessThan('$(_TargetFrameworkVersionWithoutV)', '6.0'))">$(DefaultAppHostRuntimeIdentifier.Replace("arm64", "x64"))</DefaultAppHostRuntimeIdentifier>
+    <!-- If we are running on win-arm64 and the TFM is < 5.0, we have to use a x64 apphost since there are no win-arm64 apphosts previous to .NET 5.0. -->
```
Comment (joeloff) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:140]: Didn't we back port ARM-64 to 3.1?

Comment (marcpopMSFT) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:140]: Not for public consumption.

**Comment by sfoslund** [comment]
@wli3 I've tested this manually and confirmed no msbuild typos.

**Comment by sfoslund** [comment]
@wli3 can I get a quick review here?

**Review by wli3 — APPROVED** [review]

**Pull request merged**